jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/NodeFunctions.java

```diff
@@ -172,14 +166,20 @@ public static boolean rdfTermEquals(Node n1, Node n2) {
             // Two literals, may be sameTerm by language tag case insensitivity.
             String lang1 = n1.getLiteralLanguage() ;
             String lang2 = n2.getLiteralLanguage() ;
-
-            if ( !lang1.equals("") && lang1.equalsIgnoreCase(lang2) ) {
-                // Two language tags, equal by case insensitivity.
-                boolean b = n1.getLiteralLexicalForm().equals(n2.getLiteralLexicalForm()) ;
-                if ( b )
-                    return true ;
+            if ( ! lang1.equals("") && ! lang2.equals("") ) {
```

Comment:
Can `lang1`/`lang2` ever be `null`?  Would `StringUtils.isNotBlank()` be safer (assuming we have `commons-lang` as a dependency here)?

Reply:
`StringUtils.isEmpty`. (`isBlank` includes whitespace.)

`getLiteralLanguage()` is never null. Documented on `LiteralLabel.language()`.